test/integration/lib/render.js

```diff
@@ -125,26 +140,27 @@ exports.run = function (implementation, ignores, render) {
 
             png.data = data;
 
-            if (process.env.UPDATE) {
+            let writeExpected = !!process.env.UPDATE;
+            try {
+                stats = fs.statSync(expected, fs.R_OK | fs.W_OK);
+                if (!stats.isFile()) throw new Error();
+
+            } catch (e) { // no expected.png, create it
+                writeExpected = true;
+            }
```

Comment:
I think we should keep the requirement for an explicitly setting `UPDATE` rather than always writing `expected.png` when it's missing

Reply:
I didn't change the logic, just deduplicated some code here. You can see that it creates expected.png when its missing if you don't set UPDATE.

Reply:
🙈  comment withdrawn